api/app/Models/ArticlePermalink.php

```diff
@@ -16,14 +19,39 @@
  * @package App\Models
  *
  */
-class ArticlePermalink extends BaseModel
+class ArticlePermalink extends ChildBaseModel
 {
     public $table = 'article_permalinks';
 
     protected $primaryKey = 'permalink';
 
+
+
     public function article()
     {
         return $this->belongsTo(Article::class, 'article_id', 'article_id');
     }
+
+
+
+    /**
+     * @param Builder $query
+     * @param BaseModel $parent
+     * @return Builder
+     */
+    protected function attachParentModelToQuery(Builder $query, BaseModel $parent)
+    {
+        $query->where('article_id','=',$parent->article_id);
+        return $query;
+    }
+
+
+    /**
+     * @param BaseModel $parent
+     * @return void
+     */
+    public function attachParent(BaseModel $parent)
+    {
+        $this->article_id = $parent->article_id;
+    }
```

Comment:
Having child controllers aware of a relationship will mean we don't need `ChildBaseModels` - it doesn't actually make sense for a model to be constrained to just being a child, as some items like `User` will simultaneously be a child model and a parent model (Eg `Article` `hasOne` `Author`[`User`] but `User` has it's own crud.)